File: pytools/data/data.code.lua.server/ReqSceneEnterFly.lua
module("ReqSceneEnterFly", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneEnterFly})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.map_id = 0
end


function decode(self, pack)
	self.map_id = pack:readU32()
	return self
end


function getMapId(self)
	return self.map_id
end

File: pytools/examples/golang/src/server/proto/req_scene_enter_fly.go
package proto

import (
	"packet"
)

type ReqSceneEnterFly struct {
	mapId                    uint32
}

func ReqSceneEnterFlyDecode(pack *packet.Packet) *ReqSceneEnterFly {
	reqSceneEnterFly := &ReqSceneEnterFly{}

	reqSceneEnterFly.mapId = pack.ReadUint32()
	return reqSceneEnterFly
}

func (this *ReqSceneEnterFly) GetMapId() uint32 {
	return this.mapId
}
